hydra/core/plugins.py

```diff
@@ -103,6 +106,57 @@ def is_in_toplevel_plugins_module(clazz: str) -> bool:
             "hydra._internal.core_plugins."
         )
 
+    @staticmethod
+    def _setup_plugin(
+        plugin: Any,
+        task_function: TaskFunction,
+        config: DictConfig,
+        config_loader: Optional[ConfigLoader] = None,
+        hydra_context: Optional[HydraContext] = None,
+    ) -> Any:
+        """
+        With HydraContext introduced in #1581, we need to set up the plugins in a way
+        that's compatible with both Hydra 1.0 and Hydra 1.1 syntax.
+        This method should be deleted in the next major release.
+        """
+        assert isinstance(plugin, Sweeper) or isinstance(plugin, Launcher)
+        assert (
+            config_loader is not None or hydra_context is not None
+        ), "config_loader and hydra_context cannot both be None"
+
+        param_keys = signature(plugin.setup).parameters.keys()
+
+        if "hydra_context" not in param_keys:
+            warnings.warn(
+                message=(
+                    "\n"
+                    "\tPlugin's setup() signature has changed in Hydra 1.1.\n"
+                    "\tSupport for the old style will be removed in Hydra 1.2.\n"
+                    "\tFor more info, check https://github.com/facebookresearch/hydra/pull/1581.\n"
+                ),
```

Comment:
use dedent (here and in the tests). see other places in the codebase to see exact usage.